File: src/main/java/io/github/hydrazinemc/oxidiser/event/player/PlayerSwingHandEvent.java
package io.github.hydrazinemc.oxidiser.event.player;

import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.util.Hand;
import io.github.hydrazinemc.oxidiser.event.StimulusEvent;

/**
 * Called when a {@link ServerPlayerEntity} swings their hand.
 */
public interface PlayerSwingHandEvent {
    StimulusEvent<PlayerSwingHandEvent> EVENT = StimulusEvent.create(PlayerSwingHandEvent.class, ctx -> (player, hand) -> {
        try {
            for (var listener : ctx.getListeners()) {
                listener.onSwingHand(player, hand);
            }
        } catch (Throwable t) {
            ctx.handleException(t);
        }
    });

    void onSwingHand(ServerPlayerEntity player, Hand hand);
}

File: src/main/kotlin/io/github/hydrazinemc/oxidiser/event/player/PlayerSwingHandEvent.kt
package io.github.hydrazinemc.oxidiser.event.player

import io.github.hydrazinemc.oxidiser.event.Event
import net.minecraft.server.network.ServerPlayerEntity
import net.minecraft.util.Hand

/**
 * Called when a [ServerPlayerEntity] swings their hand.
 */
object PlayerSwingHandEvent : Event<PlayerSwingHandEvent.EventData>() {
	data class EventData(val player: ServerPlayerEntity, val hand: Hand)
}